templates/contribute/style.md

```diff
@@ -565,3 +565,29 @@ Documentation strings for declarations are delimited with `/-- -/`.
 
 See our [documentation requirements](doc.html) for more suggestions
 and examples.
+
+### Deprecation
+
+Deleting, renaming, or changing declarations can cause downstreams projects
+that rely on these definitions to fail to compile. Any publicly exposed theorems and
+definitions that are being removed should be gracefully transitioned by keeping the
+old theorem statement with a `@[deprecated]` attribute, to warn downstream projects to
+transition away from deprecated definitions before they are deleted. Renamed theorems
+can use a deprecated `alias` to the new name, while in cases where the theorem is not
+being replaced, the deprecated theorem definition can be marked with an attribute, like so:
+
+```lean4
+theorem new_name : ... := ...
+@[deprecated new_name (since := "YYYY-MM-DD")] alias old_name := new_name
```

Comment:
Actually, isn't the syntax different? (Here's one random example: https://github.com/leanprover-community/mathlib4/pull/23841/files)
```suggestion
@[deprecated (since := "YYYY-MM-DD")] alias old_name := new_name

Reply:
You're right, though according to https://github.com/leanprover/lean4/pull/6112 I thought this should print a warning. It turns out that using an alias (instead of say, a def) seems to suppress the warning, though I'm not sure how.